packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2546,20 +2595,35 @@ export function checkForWrongSuspensePriorityInDEV(sourceFiber: Fiber) {
 
 function flushSuspensePriorityWarningInDEV() {
   if (__DEV__) {
-    if (componentsWithSuspendedDiscreteUpdates !== null) {
+    if (
+      componentsWithSuspendedDiscreteUpdates !== null &&
+      componentsThatCallSuspendedDiscreteUpdates.size > 0
+    ) {
       const componentNames = [];
       componentsWithSuspendedDiscreteUpdates.forEach(name => {
         componentNames.push(name);
       });
       componentsWithSuspendedDiscreteUpdates = null;
 
-      // TODO: A more helpful version of this message could include the names of
-      // the component that were updated, not the ones that suspended. To do
-      // that we'd need to track all the components that updated during this
-      // render, perhaps using the same mechanism as `markRenderEventTime`.
+      const parentComponentNamesString = Array.from(
+        componentsThatCallSuspendedDiscreteUpdates,
```

Comment:
Older browsers (and I think all versions of IE) don't support `Array.from`. That's why the code above uses a `forEach` loop instead.

In general, try to stay consistent with other parts of the codebase. But also don't be afraid to change existing code if you need to. For example, if we were to use `Array.from` here, then we would replace the `forEach` loop above, too.